### PR TITLE
ODE with Operator BC Documentation

### PR DESCRIPTION
**Pull request by ericmyzelev**

### CODE REVIEW

File: docs/demos/pinn_forward.rst [hunk 1]
```diff
@@ -40,6 +40,7 @@ Time-dependent PDEs
    pinn_forward/diffusion.1d.exactBC
    pinn_forward/diffusion.1d.resample
    pinn_forward/burgers.rar
+   pinn_forward/OperatorBC
```
Comment (lululxvi) [inline docs/demos/pinn_forward.rst:43]: This should be in the section of "ODEs".

**Comment by lululxvi** [comment]
@mitchelldaneker What is the name of the ODE?

**Comment by lululxvi** [comment]
Reformat the code via [black](https://github.com/psf/black)

File: examples/pinn_forward/Operator_BC.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch"""
```
Comment (lululxvi) [inline examples/pinn_forward/Operator_BC.py:1]: Did you test these three backends?

**Comment by mitchelldaneker** [comment]
> @mitchelldaneker What is the name of the ODE?

It is a Nonhomogeneous differential equation with linear forcing.

File: examples/pinn_forward/Operator_BC.py (hunk 2 above)
Comment (ericmyzelev) [reply examples/pinn_forward/Operator_BC.py:1]: Yes I did

File: docs/demos/pinn_forward/OperatorBC.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
+ODE in 1D with boundary conditions created with OperatorBC
```
Comment (lululxvi) [inline docs/demos/pinn_forward/OperatorBC.rst:1]: Title: Second order ODE

**Comment by lululxvi** [comment]
Please use the file names:
- ode_2nd.py
- ode.2nd.rst

File: docs/demos/pinn_forward/OperatorBC.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+=================================================================
+
+Problem setup
+--------------
+
+We will solve an ODE:
+
+.. math:: y''(t)-10y'(t)+9y(t) = 5t
+
+with boundary/initial conditions
```
Comment (lululxvi) [inline docs/demos/pinn_forward/OperatorBC.rst:11]: with boundary conditions

File: docs/demos/pinn_forward/OperatorBC.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+    geom = dde.geometry.TimeDomain(0, 0.25)
+
+Next, we express the ODE residual of the ODE:
+
+.. code-block:: python
+
+    def ode(t, y):
+    	dy_dt = dde.grad.jacobian(y, t)
+   	d2y_dt2 = dde.grad.hessian(y, t)
```
Comment (lululxvi) [inline docs/demos/pinn_forward/OperatorBC.rst:40]: The indent here is not correct. Do not use tab; always use spaces.

@mitchelldaneker You should also double check.

File: docs/demos/pinn_forward/ode.2nd.rst [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+=================================================================
+
+Problem setup
+--------------
+
+We will solve an ODE:
+
+.. math:: y''(t)-10y'(t)+9y(t) = 5t
+
+with boundary conditions
```
Comment (lululxvi) [inline docs/demos/pinn_forward/ode.2nd.rst:11]: with initial conditions

File: docs/demos/pinn_forward/ode.2nd.rst [hunk 7]
```diff
@@ -0,0 +1,103 @@
+Second order ODE
+=================================================================
```
Comment (lululxvi) [inline docs/demos/pinn_forward/ode.2nd.rst:2]: The length of === or --- should be the same as the words above.

File: docs/demos/pinn_forward/ode.2nd.rst [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    import deepxde as dde
+    import numpy as np
+
+We begin by defining a computational geometry. We can use a built-in class ``TimeDomain`` as follows
+
+.. code-block:: python
+
+    geom = dde.geometry.TimeDomain(0, 0.25)
+
+Next, we express the ODE residual of the ODE:
```
Comment (lululxvi) [inline docs/demos/pinn_forward/ode.2nd.rst:34]: express the residual of the ODE

File: docs/demos/pinn_forward/ode.2nd.rst [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    def error_2(inputs, outputs, X):
+        return dde.grad.jacobian(outputs, inputs, i=0, j=None) - 2
+
+Then, the boundary condition is defined by
+
+.. code-block:: python
+
+    bc = dde.icbc.OperatorBC(geom, error_2, boundary_l)
+
+Now, we have specified the geometry, PDE residual, the initial condition and the boundary condition. We then define the PDE problem. Note: If you use `X` in `func`, then do not set ``num_test`` when you define ``dde.data.PDE`` or ``dde.data.TimePDE``, otherwise DeepXDE would throw an error. In this case, the training points will be used for testing, and this will not affect the network training and training loss. This is a bug of DeepXDE, which cannot be fixed in an easy way for all backends.
```
Comment (lululxvi) [inline docs/demos/pinn_forward/ode.2nd.rst:73]: the initial conditions.

File: examples/pinn_forward/ode_2nd.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+data = dde.data.TimePDE(geom, ode, [bc1, bc2], 16, 2, solution=func, num_test=100)
+layer_size = [1] + [50] * 3 + [1]
+activation = "tanh"
+initializer = "Glorot uniform"
+net = dde.nn.FNN(layer_size, activation, initializer)
+
+model = dde.Model(data, net)
+
+# model.compile("adam", lr=.01, metrics=["l2 relative error"])
```
Comment (lululxvi) [inline examples/pinn_forward/ode_2nd.py:43]: Delete these two commented lines.

File: examples/pinn_forward/ode_2nd.py [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+def bc_func1(inputs, outputs, X):
+    return outputs + 1
+
+
+def bc_func2(inputs, outputs, X):
+    return dde.grad.jacobian(outputs, inputs, i=0, j=None) - 2
+
+
+bc1 = dde.icbc.OperatorBC(geom, bc_func1, boundary_l)
```
Comment (lululxvi) [inline examples/pinn_forward/ode_2nd.py:31]: For BC 1, you need to use `IC`. No need to use `OperatorBC`.

File: docs/demos/pinn_forward/ode.2nd.rst [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+The first argument to ``ode`` is the network input, i.e., the :math:`t`-coordinate. The second argument is the network output, i.e., the solution :math:`y(t)`, but here we use ``y`` as the name of the variable.
+
+We define the initial condition, setting the value of the function at :math:`t=0` to -1.
+   
+.. code-block:: python
+
+    ic = dde.icbc.IC(geom, lambda x: -1, lambda _, on_initial: on_initial)
+    
+Now we deal with the boundary condition :math:`y'(0)=2`.
```
Comment (lululxvi) [inline docs/demos/pinn_forward/ode.2nd.rst:51]: boundary condition --> initial condition

Also change in other places.

**Pull request merged**